repl/src/main/resources/fake_shell.py

```diff
@@ -534,8 +534,20 @@ def main():
             exec('from pyspark.sql import HiveContext', global_dict)
             exec('from pyspark.streaming import StreamingContext', global_dict)
             exec('import pyspark.cloudpickle as cloudpickle', global_dict)
+
             if spark_major_version >= "2":
                 exec('from pyspark.shell import spark', global_dict)
+            else:
+                # LIVY-294, need to check whether HiveContext can work properly,
+                # fallback to SQLContext if HiveContext can not be initialized successfully.
+                # Only for spark-1.
+                code = """import py4j
+from pyspark.sql import SQLContext
+try:
+  sqlContext.tables()
+except py4j.protocol.Py4JError:
+  sqlContext = SQLContext(sc)"""
```

Comment:
didn't find a right approach. Previous I thought the following approach would work, but unfortunately, it doesn't

```
                code = """import py4j
                             ...:from pyspark.sql import SQLContext
                             ...:try:
                             ...:  sqlContext.tables()
                             ...:except py4j.protocol.Py4JError:
                             ...:  sqlContext = SQLContext(sc)"""
```